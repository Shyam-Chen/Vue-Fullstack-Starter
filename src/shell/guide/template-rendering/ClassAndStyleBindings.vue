<template>
  <v-layout column class="my-1">
    <div class="title mb-2">Class and Style Bindings</div>

    <div :class="{ ['cyan-color']: activeCyan }">Cyan</div>
    <div :class="pink">Pink</div>
    <div :class="[amber]">Amber</div>

    <div><span :style="{ color: teal800, backgroundColor: teal100 }">Teal</span></div>
    <div><span :style="purple">Purple</span></div>
    <div><span :style="[orange800, orange100]">Orange</span></div>
  </v-layout>
</template>

<script>
// @flow

type ClassBindings = {
  activeCyan: boolean,
  pink: { 'pink-color': boolean },
  amber: string,
};

type StyleBindings = {
  teal800: string,
  teal100: string,
  purple: { color: string, backgroundColor: string },
  orange800: { color: string },
  orange100: { backgroundColor: string },
};

type Data = ClassBindings & StyleBindings;

export default {
  data(): Data {
    return {
      // class bindings
      activeCyan: true,
      pink: {
        'pink-color': true,
      },
      amber: 'amber-color',

      // style bindings
      teal800: '#00695C',
      teal100: '#B2DFDB',
      purple: {
        color: '#6A1B9A',  // purple 800
        backgroundColor: '#E1BEE7',  // purple 100
      },
      orange800: {
        color: '#EF6C00',
      },
      orange100: {
        backgroundColor: '#FFE0B2',
      },
    };
  },
};
</script>

<style scoped>
.cyan-color {
  color: #00bcd4;
}

.pink-color {
  color: #e91e63;
}

.amber-color {
  color: #ffc107;
}
</style>
