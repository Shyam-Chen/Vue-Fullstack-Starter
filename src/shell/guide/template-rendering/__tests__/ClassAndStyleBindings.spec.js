import { shallowMount } from '@vue/test-utils';

import ClassAndStyleBindings from '../ClassAndStyleBindings';

describe('ClassAndStyleBindings', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(ClassAndStyleBindings);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
