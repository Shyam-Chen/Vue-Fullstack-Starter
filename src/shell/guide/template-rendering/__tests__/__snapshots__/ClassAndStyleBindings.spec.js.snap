// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassAndStyleBindings should render an initial component 1`] = `
<v-layout column="" class="my-1">
  <div class="title mb-2">Class and Style Bindings</div>
  <div class="cyan-color">Cyan</div>
  <div class="pink-color">Pink</div>
  <div class="amber-color">Amber</div>
  <div><span style="color: rgb(0, 105, 92); background-color: rgb(178, 223, 219);">Teal</span></div>
  <div><span style="color: rgb(106, 27, 154); background-color: rgb(225, 190, 231);">Purple</span></div>
  <div><span style="color: rgb(239, 108, 0); background-color: rgb(255, 224, 178);">Orange</span></div>
</v-layout>
`;
